### DISCUSSION TITLE
Additional sensor data for charging time

### DISCUSSION THREAD

**Discussion by nisnu** [marq24/ha-evcc#28] [Ideas]
If I have an active charging plan the hours of less expensive hours are calculated by EVCC to charge at that hours. To be able to forecast the engery consumption of the house upfront (e.g. to be able charging sufficiently the house battery from grid) it would be great to have a sensor displaying the hours of when charging is planned.

**Comment by marq24** [comment]
well - where there is an value available in evcc API then this could be added - but this integration can't invent/create new sensors... [or can you provide the "Math" for the new sensors?

**Reply by RenatusRo** [reply]
doesn't it come with the api/loadpoints/xx/plan ?

`{"result":{"planTime":"2024-10-17T16:15:00Z","duration":4869,"plan":[{"start":"2024-10-17T14:22:53.403123789+02:00","end":"2024-10-17T13:00:00Z","price":0.25457013},{"start":"2024-10-17T13:00:00Z","end":"2024-10-17T13:44:02.841384658Z","price":0.26015633}],"power":460}}
`

**Reply by marq24** [reply]
well the loadpoint plantime sensor is already there - it just need to be enabled by the user...

**Reply by nisnu** [reply]
Thanks for the reply. Actually I can see only the sensor evcc_garage_plan_time with a planed time but this is the time I entered when I want to leave. E.g. this could be tomorrow 7:00. but because the price is cheaper before the loading time could be between 1:00 and 3:00.
Do you mean a different sensor?
I am not so technical guy so I don't know what the API offers, sorry for that.
